addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/handler/YamahaZoneThingHandler.java

```diff
@@ -0,0 +1,606 @@
+        //
+        // List<StateOption> options = new ArrayList<StateOption>();
+        // for (String inputName : msg.available_inputs) {
+        // options.add(new StateOption(inputName, inputName));
+        // }
+        //
+        // StateDescription state = null;
+        // state = new StateDescription(null, null, null, "%s", false, options);
+        //
+        // ChannelType type = new ChannelType(channel.getChannelTypeUID(), false, "String", channel.getKind(),
+        // channel.getLabel(), channel.getDescription(), null, channel.getDefaultTags(), state, null, null);
+
+        // TODO Use a ChannelType provider to add this changed channel type
+        /**
+         * TODO Channel State created dynamically like in
+         * /home/david/Programming/openhab2-master2/git/openhab2-addons/addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/type/MetadataUtils.java
+         * /home/david/Programming/openhab2-master2/git/openhab2-addons/addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/type/HomematicTypeGeneratorImpl.java
+         */
+
+        // channel = ChannelBuilder.create(channelUID, channel.getAcceptedItemType()).withType(type.getUID()).build();
+        // Thing newThing = editThing().withoutChannel(channel.getUID()).withChannel(channel).build();
+        // updateThing(newThing);
+    }
+
+    private String grpPlayback(String channelIDWithoutGroup) {
+        return new ChannelUID(thing.getUID(), YamahaReceiverBindingConstants.CHANNEL_GROUP_PLAYBACK,
+                channelIDWithoutGroup).getId();
+    }
+
+    private String grpNav(String channelIDWithoutGroup) {
+        return new ChannelUID(thing.getUID(), YamahaReceiverBindingConstants.CHANNEL_GROUP_NAVIGATION,
+                channelIDWithoutGroup).getId();
+    }
+
+    private String grpZone(String channelIDWithoutGroup) {
+        return new ChannelUID(thing.getUID(), YamahaReceiverBindingConstants.CHANNEL_GROUP_ZONE, channelIDWithoutGroup)
+                .getId();
+    }
+
+    @Override
+    public void playInfoUpdated(PlayInfoState msg) {
+        playInfoState = msg;
+        updateState(grpPlayback(YamahaReceiverBindingConstants.CHANNEL_PLAYBACK), new StringType(msg.playbackMode));
+        updateState(grpPlayback(YamahaReceiverBindingConstants.CHANNEL_PLAYBACK_STATION), new StringType(msg.Station));
+        updateState(grpPlayback(YamahaReceiverBindingConstants.CHANNEL_PLAYBACK_ARTIST), new StringType(msg.Artist));
+        updateState(grpPlayback(YamahaReceiverBindingConstants.CHANNEL_PLAYBACK_ALBUM), new StringType(msg.Album));
+        updateState(grpPlayback(YamahaReceiverBindingConstants.CHANNEL_PLAYBACK_SONG), new StringType(msg.Song));
+    }
+
+    @Override
+    public void playControlUpdated(PlayControlState msg) {
+        playControlState = msg;
+        // TODO show preset name. The channel state options have to be changed for that
```

Comment:
Is there a developer documentation of how to dynamically change channel states that are statically defined in the xml? As far as I have found out, it's not straight forward and other addons have plenty of boiler code to archive this. I'd like to avoid another hundreds of lines for this merge request to be honest.